src/hooks/useConnection.tsx

```diff
@@ -158,27 +141,22 @@ export function useConnection() {
   }
 
   const [
-    { provider, onboard, signer, network, address, error, isConnected },
+    { provider, onboard, signer, chainId, account, error, isConnected },
     dispatch,
   ] = context;
   const connect = React.useCallback(async () => {
     try {
       const onboardInstance = Onboard({
-        dappId: config(network).onboardConfig.apiKey,
-        hideBranding: true,
-        networkId: 1, // Default to main net. If on a different network will change with the subscription.
+        ...onboardBaseConfig(chainId as ValidChainId),
         subscriptions: {
-          address: (address: string | null) => {
-            dispatch({ type: "set address", address });
+          address: (address: string) => {
+            dispatch({ type: "set account", account: address });
           },
           network: async (networkId) => {
-            onboard?.config({ networkId: networkId });
+            onboard?.config({ networkId });
             dispatch({
-              type: "set network",
-              network: {
-                chainId: networkId,
-                name: getNetworkName(networkId as ChainId),
-              },
+              type: "set chainId",
+              chainId: networkId as ValidChainId,
```

Comment:
I think we need a comment here at least on why we are treating this as chainId

Reply:
Will address this in the next PR!